### PR TITLE
Add access url for sub organization console and myaccount apps

### PR DESCRIPTION
**Pull request by ChanikaRuchini**
## Purpose
-  When creating a sub-organization, console and my account applications are shared, Added the relevant access URL for the  apps when sharing.

## Related Issue
- https://github.com/wso2/product-is/issues/18933

### CODE REVIEW

File: components/org.wso2.carbon.identity.organization.management.application/src/main/java/org/wso2/carbon/identity/organization/management/application/OrgApplicationManagerImpl.java [hunk 1]
```diff
@@ -602,7 +603,8 @@ public void shareApplication(String ownerOrgId, String sharedOrgId, ServiceProvi
             }
             String sharedApplicationId;
             try {
-                ServiceProvider delegatedApplication = prepareSharedApplication(mainApplication, createdOAuthApp);
+                ServiceProvider delegatedApplication =
+                        prepareSharedApplication(mainApplication, createdOAuthApp, ownerOrgId, sharedTenantDomain);
```
Comment (sadilchamishka) [inline components/org.wso2.carbon.identity.organization.management.application/src/main/java/org/wso2/carbon/identity/organization/management/application/OrgApplicationManagerImpl.java:607]:
```suggestion
                ServiceProvider delegatedApplication =
                        prepareSharedApplication(mainApplication, createdOAuthApp, sharedOrgId);
```
According to the logic, additionally we requires mainApp's tenant domain, which can be take from `mainApplication` itself, and the shared-org-id which need to be added after /o/<org-id>. Hence we can pass the sharedOrgId instead the `sharedTenantDomain` though they seems equal.

Comment (ChanikaRuchini) [reply components/org.wso2.carbon.identity.organization.management.application/src/main/java/org/wso2/carbon/identity/organization/management/application/OrgApplicationManagerImpl.java]: Fixed

File: components/org.wso2.carbon.identity.organization.management.application/src/main/java/org/wso2/carbon/identity/organization/management/application/OrgApplicationManagerImpl.java [hunk 2]
```diff
@@ -755,11 +758,23 @@ private ServiceProvider prepareSharedApplication(ServiceProvider mainApplication
         delegatedApplication.setApplicationName(oAuthConsumerApp.getApplicationName());
         delegatedApplication.setDescription(mainApplication.getDescription());
         delegatedApplication.setInboundAuthenticationConfig(inboundAuthConfig);
+        if (FrameworkConstants.Application.CONSOLE_APP.equals(mainApplication.getApplicationName())) {
```
Comment (dewniMW) [inline components/org.wso2.carbon.identity.organization.management.application/src/main/java/org/wso2/carbon/identity/organization/management/application/OrgApplicationManagerImpl.java:761]: In ApplicationMgtUtil there are utils to check whether it is console / my account. we can reuse those

File: components/org.wso2.carbon.identity.organization.management.application/src/main/java/org/wso2/carbon/identity/organization/management/application/OrgApplicationManagerImpl.java [hunk 3]
```diff
@@ -755,11 +758,25 @@ private ServiceProvider prepareSharedApplication(ServiceProvider mainApplication
...
+        }
         appendFragmentAppProperties(delegatedApplication);
 
         return delegatedApplication;
     }
 
+    private String resolveAccessURL(String ownerTenantDomain, String sharedOrgId, String appPath) {
+
+        String accessUrl = IdentityUtil.getServerURL(appPath, true, true);
+        accessUrl = accessUrl.replace(appPath, "/t/" + ownerTenantDomain + "/o/" + sharedOrgId + appPath);
```
Comment (dewniMW) [inline components/org.wso2.carbon.identity.organization.management.application/src/main/java/org/wso2/carbon/identity/organization/management/application/OrgApplicationManagerImpl.java:776]: use the FrameworkConstants for /t/ and /o/

Comment (ChanikaRuchini) [reply components/org.wso2.carbon.identity.organization.management.application/src/main/java/org/wso2/carbon/identity/organization/management/application/OrgApplicationManagerImpl.java]: Fixed

**Pull request merged**